src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2770,6 +2827,9 @@ def set_up_workload_auto_scaler_profile(self, mc: ManagedCluster) -> ManagedClus
                 self.models.ManagedClusterWorkloadAutoScalerProfileKeda(enabled=True)  # pylint: disable=no-member
             )
 
+        # set up addon autoscaling
+        mc = self.set_up_addon_autoscaling(mc)
```

Comment:
Better not running another set up profile func in `set_up_workload_auto_scaler_profile`, please run it in the function responsible for constructing mc profile. OR

If the new "set up" does something similar to the existing "set up", you can add the new changes to the current function.

Reply:
update!